### PR TITLE
Fix: colon on last position in line (Label)

### LINKED ISSUE #161: Wrong indent, if colon present as last character in command
### Description

"Statements in functions are indented one level more than expected."

### Unformatted input snippet

```autohotkey
foo()
{
if (true) {
for i, drive in drives {
DriveGet, totalSpace, Capacity, %drive%:
DriveSpaceFree, freeSpace, %drive%:
GuiControl, Show, Drive%i% ; must be NOT indented
}
} else {
bar()
}
}
```

### Expected formatted output snippet

```autohotkey
foo()
{
    if (true) {
        for i, drive in drives {
            DriveGet, totalSpace, Capacity, %drive%:
            DriveSpaceFree, freeSpace, %drive%:
            GuiControl, Show, Drive%i%
        }
    } else {
        bar()
    }
}
```

### Actual formatted output snippet

```autohotkey
foo()
{
    if (true) {
        for i, drive in drives {
            DriveGet, totalSpace, Capacity, %drive%:
            DriveSpaceFree, freeSpace, %drive%:
                GuiControl, Show, Drive%i% ; must be NOT indented
            }
        } else {
            bar()
        }
    }
```

### Additional context

Working on fixing this issue.

### PR DESCRIPTION
**Pull request by kyklish**
Closes #161.

Changes proposed in this pull request:

-   check label name on presence forbidden characters

Notifying @mark-wiemer

### CODE REVIEW

File: src/providers/formattingProvider.ts [hunk 1]
```diff
@@ -209,8 +209,11 @@ export class FormatProvider implements vscode.DocumentFormattingEditProvider {
                 depth++;
             }
 
-            // default or hotkey
-            if (!moreOpenParens && purifiedLine.match(/:\s*$/)) {
+            // default or hotkey or label
+            if (
+                !moreOpenParens &&
+                purifiedLine.match(/^(?:\s*[^\s\t,`]+):\s*$/) // Label name may consist of any characters other than space, tab, comma and the escape character (`)
```
Comment (mark-wiemer) [inline src/providers/formattingProvider.ts:215]: Same as https://github.com/mark-wiemer/vscode-autohotkey-plus-plus/pull/163#discussion_r922964092, let's add some unit tests to make this regex a bit more readable.

**Review by mark-wiemer — CHANGES_REQUESTED** [review]

File: src/providers/formattingProvider.ts (hunk 1 above)
Comment (kyklish) [reply src/providers/formattingProvider.ts]: Done.

**Review by mark-wiemer — DISMISSED** [review]
You're quite good at this :)

**Review by mark-wiemer — APPROVED** [review]

**Comment by mark-wiemer** [comment]
Can you help resolve conflicts here, @kyklish ? Then we can finally merge this!

**Comment by kyklish** [comment]
Done.

**Comment by kyklish** [comment]
Definitely done :)

**Pull request merged**